caiman/motion_correction.py

```diff
@@ -66,6 +66,14 @@
 from cv2 import dft as fftn
 from cv2 import idft as ifftn
 opencv = True
+try:
```

Comment:
With has_cuda and use_cuda as the way this works, this means:
1) If the user doesn't have cuda installed, their code will do the right thing even if they request cuda; it'll just ignore their request
2) If the user has cuda installed and they request cuda, it will use cuda
3) If the user has cuda installed and they request cuda but it doesn't work (can't get the device, or something goes wrong initialising it because of dependency issues, or ...) then they're in for some messiness

I think this is probably a fine approach, with the alternative that comes immediately to mind to try to initialise cuda if it is requested, and if it fails fallback to the non-cuda algorithms transparently as in case 1 above.

Reply:
Getting cuda to run with multiprocessing was difficult. The order of imports matters (hence the delayed import of skcuda) and the device context needs to be at global scope. I'd therefore advise against delayed initialization of cuda as it may break multiprocessing compatibility. However, there's now a message in case the user selects CUDA although it is unavailable.